flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -508,6 +515,10 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 					errors.BadSpecificationError, fmt.Sprintf("failed to find ArrayNode subNode task with id: '%s'", taskID), nil)), nil
 			}
 
+			runtimeData := taskNode.CoreTask().GetMetadata().GetRuntime()
+			// check if the literal can be offloaded based on the runtime data and the offloading config
+			canOffloadLiteral = a.literalOffloadingConfig.Enabled && a.literalOffloadingConfig.IsSupportedSDKVersion(runtimeData.GetType().String(), runtimeData.GetVersion())
```

Comment:
nit, maybe define this closer to where it's used?

Reply:
+1

Reply:
The reason being,this info is only available under taskMetadata and hence i moved it inside the case statement instead of introducing another conditional to check the same a few lines below
```
switch arrayNode.GetSubNodeSpec().GetKind() {
		case v1alpha1.NodeKindTask:
```
Let me know what you guyz think.

Reply:
I'm OK with whatever you want. If this stops adding a different switch just to get this to work that's fine.